### PR TITLE
feat: add docs for compaction offload mechanism

### PR DESCRIPTION
**Pull request by LeslieKid**
### Tasks
Add tech-design doc for compaction offloading.

### CODE REVIEW

**Comment by jiacai2050** [comment]
The CI failed, please execute `make lint` before submit.

File: content/en/docs/design/compaction_offload.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+The procedure of compaction based above architecture diagram is:
+
+1. HoraeDB triggers compaction procedure under some conditions and then generates the compaction task.
+2. HoraeDB fetchs the information of suitable compaction node from the HoraeMeta.
+3. HoraeDB distributes the compaction task to the remote compaction node, according to the information fetch from HoraeMeta.
+4. Compaction node executes the task and send the result back to the HoraeDB.
+5. HoraeDB receives the result and updates the manifest.
+
+We can see that the compaction offload architecture in HoraeDB is different from the traditional one. It separates the compaction task distribution from HoraeMeta and moves it into HoraeDB, which reduces the load of HoraeMeta to lower the risk of single points of failure.
```
Comment (jiacai2050) [inline content/en/docs/design/compaction_offload.md:50]: > We can see that the compaction offload architecture in HoraeDB is different from the traditional one.

I think most system will do like this, only the horaedb node know which sst it has, the meta have no idea how many sst a table contains, so meta can't the schedule task.

Comment (LeslieKid) [reply content/en/docs/design/compaction_offload.md]: Thanks @jiacai2050 . The statement here is indeed incorrect.

fixed.

File: content/en/docs/design/compaction_offload.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+## Overview
+
+```plaintext
+┌───────────────────────────────────────────────────────────────────────┐
+│                                                                       │
+│                           HoraeMeta Cluster                           │
+│                                                                       │
+└───────────────────────────────────────────────────────────────────────┘
+    ▲                ▲                       |                   |
```
[the remark this answers is not in the document]
Comment (LeslieKid) [reply content/en/docs/design/compaction_offload.md]: fixed.

File: content/en/docs/design/compaction_offload.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+```
+
+The diagram above describes the architecture of cluster for compaction offload, where some key concepts need to be explained:
+
+- `Compaction Node`: Takes responsibility to handle offloaded compaction tasks. The compaction node receives the compaction task and performs the actual merging of SSTables, then sends back the task result to HoraeDB.
+- `HoraeMeta Cluster`: HoraeMeta acts as a compaction nodes manager in the compaction offload scenario. It monitors the compaction nodes cluster and schedule the compaction nodes.
+
+The procedure of remote compaction based above architecture diagram is:
+
+1. HoraeDB fetchs the information of suitable compaction node from the HoraeMeta.
```
Comment (jiacai2050) [inline content/en/docs/design/compaction_offload.md:44]:
```suggestion
1. HoraeDB servers fetch the information of suitable compaction nodes from the HoraeMeta.
```

File: content/en/docs/design/compaction_offload.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+2. HoraeDB distributes the compaction task to the remote compaction node, according to the information fetch from HoraeMeta.
```
Comment (jiacai2050) [inline content/en/docs/design/compaction_offload.md:45]:
```suggestion
2. HoraeDB submit the compaction task to the remote compaction node, according to the information fetched from HoraeMeta.
```

File: content/en/docs/design/compaction_offload.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+3. Compaction node executes the task and outputs compaction results to the temporary workspace.
```
Comment (jiacai2050) [inline content/en/docs/design/compaction_offload.md:46]:
```suggestion
3. Compaction node executes the task and write results to the temporary workspace.
```

**Review by jiacai2050 — APPROVED** [review]
LGTM

**Pull request merged**